test/common/http/conn_manager_impl_test.cc

```diff
@@ -385,6 +387,10 @@ TEST_F(HttpConnectionManagerImplTest, StartAndFinishSpanNormalFlow) {
         callbacks.addStreamDecoderFilter(filter);
       }));
 
+  // Treat request as internal, otherwise x-request-id header will be overwritten.
```

Comment:
When request is considered an edge request, then its `x-request-id` header is going to be ignored and overwritten.

Previously, this test case was passing by accident, because `random_.uuid()` wasn't mocked, so even though this test case exercised the code path to overwrite the header, `random_.uuid()` returned empty value, so `if (!uuid.empty())` condition (removed in this PR) wasn't met and header wasn't overwritten.

Having said that, after I've added default response to the mock and `ASSERT()` no longer fires, this change could be removed from 2 of the 4 test cases.